custom/_legacy/mvp_docs/pillows.py

```diff
@@ -149,17 +149,22 @@ def process_indicators(self, namespaces, domain, doc_dict):
 
         for xform_id in xform_ids:
             try:
-                xform_dict = XFormInstance.get_db().get(xform_id)
-                xform_doc = IndicatorXForm.wrap_for_indicator_db(xform_dict)
+                # first try to get the doc from the indicator DB
+                xform_doc = IndicatorXForm.get(xform_id)
             except ResourceNotFound:
-                pillow_eval_logging.error(
-                    "Could not find an XFormInstance with id %(xform_id)s "
-                    "related to Case %(case_id)s" % {
-                        'xform_id': xform_id,
-                        'case_id': doc_dict['_id'],
-                    }
-                )
-                continue
+                # if that fails fall back to the main DB
```

Comment:
hmm, yeah it does.

I took this route (as opposed to loading up all the indicators) for performance reasons (many indicators are slow and load other documents), but you're right there are problems with it.

let me think about it / tweak it a bit. sorry was just excited that it worked.